azurerm/helpers/azure/servicebus.go

```diff
@@ -35,7 +35,7 @@ func ValidateServiceBusSubscriptionName() schema.SchemaValidateFunc {
 
 func ValidateServiceBusTopicName() schema.SchemaValidateFunc {
 	return validation.StringMatch(
-		regexp.MustCompile("^[a-zA-Z][-._a-zA-Z0-9]{0,258}([a-zA-Z0-9])?$"),
+		regexp.MustCompile("^[a-zA-Z][-._~a-zA-Z0-9]{0,258}([a-zA-Z0-9])?$"),
 		"The topic name can contain only letters, numbers, periods, hyphens and underscores. The namespace must start with a letter, and it must end with a letter or number and be less then 260 characters long.",
```

Comment:
Do we want to add a comment that it also supports tildes?

Reply:
Indeed we do! thanks for spotting that.